ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -697,7 +697,20 @@ def bind_ovs_ports(self, br_name, dut_iface, injected_iface, vm_iface, disconnec
             VMTopology.cmd("ovs-ofctl add-flow %s table=0,in_port=%s,action=output:%s" % (br_name, dut_iface_id, injected_iface_id))
         else:
             # Add flow from external iface to a VM and a ptf container
-            VMTopology.cmd("ovs-ofctl add-flow %s table=0,in_port=%s,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            # Allow BGP, ICMP, SNMP packets and layer2 packets from DUT to neighbors
+            # Block other traffic from DUT to EOS for EOS's stability,
+            # Allow all traffic from DUT to PTF.
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp,in_port=%s,tp_src=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp,in_port=%s,tp_dst=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp6,in_port=%s,tp_src=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp6,in_port=%s,tp_dst=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=8,icmp,in_port=%s,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=8,icmp6,in_port=%s,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
```

Comment:
For the ICMP packet, on dual-tor, there are lots of heartbeat from tors to servers, can we use the dest IP to figure out those pkts and just send them to ptf?

Reply:
Yes, more rules more accurate. At the beginning, I tried to use neighbors' IP as dst IP to filter packet, but it's not very convenient to get all neighbors' IP at ADD_TOPO phase and more rules will cost much more time to filter packets in ovs-bridge. So finally, choose protocol as category, actually, current openflow rules are a little bit complex already. We want to keep it simple and functional, not sure if there is any better tidy rules to do the same thing.